tools/src/launcher/Main.hs

```diff
@@ -517,11 +524,63 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
-    logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
+    customLogger LInfo NWallet "Starting the wallet"
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            createDirectoryIfMissing True (directory lp)
+            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            let cr = createProcHandle path args logHandle
+            system' phvar cr mempty
+        Nothing ->
+           if shouldLog then
+               liftIO $ do
+               let cr = createProcInherit path args
+               system' phvar cr mempty
+           else
+              liftIO $ do
+              (_, _, _, pid) <- runInteractiveProcess path (map toString args) Nothing Nothing
+              waitForProcess pid
+
```

Comment:
Well those three functions yield `CreateProcess` which later `system'` spawns with 
`(m, stdO, stdE, ph) <- Process.createProcess p`  - this function also takes care of terminating the process and 
prints from handle. I don't know if I can replace it since It would mean I would have to change some things and my task was just to add logging so I assumed all works as expected except logging.